### DISCUSSION TITLE
Assistant: Per-Project /default context configuration

### DISCUSSION THREAD

**Discussion by NiloCK** [zed-industries/zed#22848] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

I find myself repeatedly loading the same or similar context when working in certain projects. One repeated case for me is that models always assume modern(ish) stacks, and make recommendations that aren't applicable in practice.

It would be great if zed could find a project-default context in, say, `./.zed/assistant/default.prompt`. It'd be greater-still if the project-configured context could do things like use slash commands for relative imports.

```prompt
The project here is using dated tooling. Unless I bring it up specifically as a goal, please assume that we are locked with these versions.

If it's extremely relevant, you can mention a tooling or dependency update.

/file package.json
/file vue.config.js 
/file lerna.squerna
/file gulp.js
```

### Zed Version and System Specs

Zed: v0.165.4 (Zed)
OS: Linux Wayland ubuntu 22.04
Memory: 15.4 GiB
Architecture: x86_64
GPU: Intel(R) UHD Graphics (CML GT2) || Intel open-source Mesa driver || Mesa 23.2.1-1ubuntu3.1~22.04.2

### If applicable, add mockups / screenshots to help present your vision of the feature

In the assistant panel, a fresh context could display that both the normal default and project default have been loaded, perhaps with a differentiated icon.

![image](https://github.com/user-attachments/assets/703485b3-af00-4ed5-aeee-c0740a401cf8)

Loading of unfamiliar project prompts should probably require a user confirmation (avoiding, eg, jailbreak injection or whatever)

**Comment by AndreiPashkin** [comment]
I would add that per-project prompts are also useful. 

Such a feature exists in Cursor. I'd like to add a brief description of the project, it's structure, etc for each project I'm working on - to increase quality of answers from the LLMs.
